File: src/ObjectManager/Object.Tes/FilePacks/Records/300-SSCR.Start Script.cs
using OA.Core;

namespace OA.Tes.FilePacks.Records
{
    public class SSCRRecord : Record
    {
        public override string ToString() => $"SSCR: {EDID.Value}";
        public STRVField EDID { get; set; } // Editor ID
        public STRVField DATA; // Digits

        public override bool CreateField(UnityBinaryReader r, GameFormatId format, string type, int dataSize)
        {
            if (format == GameFormatId.TES3)
                switch (type)
                {
                    case "NAME": EDID = r.ReadSTRV(dataSize); return true;
                    case "DATA": DATA = r.ReadSTRV(dataSize); return true;
                    default: return false;
                }
            return false;
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/300-SSCR.Start Script.swift
//
//  SSCRRecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class SSCRRecord: Record {
    public override var description: String { return "SSCR: \(EDID)" }
    public var EDID: STRVField = STRVField_empty  // Editor ID
    public var DATA: STRVField! // Digits
    
    override func createField(_ r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        guard format == .TES3 else {
            return false
        }
        switch type {
        case "NAME": EDID = r.readSTRV(dataSize)
        case "DATA": DATA = r.readSTRV(dataSize)
        default: return false
        }
        return true
    }
}
